### DISCUSSION TITLE
Hold Temperatur after print finish

### DISCUSSION THREAD

**Discussion by tome9111991** [Jyers/Marlin#453] [Q&A] [answered]
At the end, I told my gcode that the nozzle should be kept at 200 and the bed at 55 degrees.

 but when the finish dialog appears, the ender sets the temperature to 0 degrees. 

how can he use the temperature of my gcode?

**Comment by aryajuanda** [comment, marked as answer]
you can set it in your slicer... at End-GCODE

**Reply by tome9111991** [reply]
I already have 

When the print is finished and show the finish promt , the ender still sets the temperature to 0

**Reply by aryajuanda** [reply]
<img width="341" alt="Screen Shot 2021-04-11 at 19 42 43" src="https://user-images.githubusercontent.com/4106760/114304582-2949b880-9afe-11eb-8724-0d19bad49e02.png">

i guess its feature in the firmware.. it stop all heater n fan

**Reply by Jyers** [reply]
This is the second mention I've gotten of this. I may end up removing it to give the user more control.

**Reply by tome9111991** [reply]
normally at the end of the gcode it says that the temperatures are set to 0

in stock marlin is the same problem

**Reply by tome9111991** [reply]
Just an idea.
Create in configuration.h
a define 
For exemple : use temerature from gcode after print done

So user can choose what he want
Gcode temperature or temperature controll from firmware after print

**Reply by cruzer619** [reply]
are you using Octoprint??  I used gcode scripts to get around it.

https://github.com/Jyers/Marlin/discussions/366#discussion-3303370

**Reply by tome9111991** [reply]
No,

**Reply by Jyers** [reply]
I'll see what I can do, but if this behavior is already coded into marlin, I can't do much about it. If its a problem with base marlin, I recommend opening an issue on the upstream branch.
